src/usexp/atom.mli

```diff
@@ -0,0 +1,15 @@
+type t = private A of string [@@unboxed]
+
+type syntax = Jbuild | Dune
+
+val is_valid_dune : string -> bool
+
+val of_string : string -> t
+
+val to_string : t -> syntax -> string
```

Comment:
I don't think `to_string` should do the validation. `to_string` should just return the contents of the atom. For instance I think that's why the build of camomile is broken in travis.

The validation should be done by a separate printing function.